unattended_installer/passwords_tool/passwordsMain.sh

```diff
@@ -240,37 +221,39 @@ function main() {
             passwords_generatePassword
         fi
 
-        if [ -n "${changeall}" ]; then
+        if [ -n "${changeall}" ] || [ -n "${p_file}" ]; then
             passwords_readUsers
-            passwords_generatePassword
+            if [ -n "${adminUser}" ] && [ -n "${adminPassword}" ]; then
+                passwords_getApiToken
+                passwords_getApiUsers
+                passwords_getApiIds
+            else
```

Comment:
Death code, else never visited.

Reply:
The code in the else clause is executed when the admin passwords are not provided.